llpc/context/llpcCompiler.cpp

```diff
@@ -208,6 +208,65 @@ static void fatalErrorHandler(void *userData, const std::string &reason, bool ge
 #endif
 }
 
+// =====================================================================================================================
+// Returns the cache accessor object resulting from checking the caches for the glue shader for the given identifier.
+//
+// @param glueShaderIdentifier : The linker object for which the glue shaders are needed.
+// @param context : The context that contains the application caches.
+// @param compiler : The compiler object that contains the internal caches.
+static CacheAccessor checkCacheForGlueShader(StringRef glueShaderIdentifier, Context *context, Compiler *compiler) {
+  Hash glueShaderCacheHash =
+      PipelineDumper::generateHashForGlueShader({glueShaderIdentifier.size(), glueShaderIdentifier.data()});
+  return CacheAccessor(context, glueShaderCacheHash, compiler);
+}
+
+// =====================================================================================================================
+// Set the glue shader at glueIndex in the ELF linking with the data in the cache.  The data must be in the cache.
+//
+// @param elfLinker : The linker object for which the glue shaders are needed.
+// @param glueIndex : The index of the glue shader to be set.
+// @param cacheAccessor : The cache accessor that resulted from checking the caches for the glue shader.
+static void setGlueBinaryBlobFromCacheData(ElfLinker *elfLinker, uint32_t glueIndex,
+                                           const CacheAccessor &cacheAccessor) {
+  assert(cacheAccessor.isInCache());
+  BinaryData elf = cacheAccessor.getElfFromCache();
+  elfLinker->addGlue(glueIndex, StringRef(reinterpret_cast<const char *>(elf.pCode), elf.codeSize));
+}
+
+// =====================================================================================================================
+// Set the data in the cache to the given data.
+//
+// @param cacheAccessor : The cache accessor for the entry to be updated.
+// @param elfData : The data to use to update the cache.
+static void updateCache(CacheAccessor &cacheAccessor, const StringRef &elfData) {
+  BinaryData elfBin = {elfData.size(), elfData.data()};
+  cacheAccessor.setElfInCache(elfBin);
+}
+
+// =====================================================================================================================
+// Sets all of the glue shaders in elfLinker by getting the binary from the cache or compiling it.
+//
+// @param elfLinker : The linker object for which the glue shaders are needed.
+// @param context : The context that contains the application caches.
+// @param compiler : The compiler object that contains the internal caches.
+static void setGlueBinaryBlobsInLinker(ElfLinker *elfLinker, Context *context, Compiler *compiler) {
+  ArrayRef<StringRef> glueShaderIdentifiers = elfLinker->getGlueInfo();
+  for (uint32_t i = 0; i < glueShaderIdentifiers.size(); ++i) {
+    LLPC_OUTS("Id for glue shader" << i << ": " << llvm::toHex(glueShaderIdentifiers[i]) << "\n");
```

Comment:
Id -> ID
Also, please remove "llvm::" prefix in cpp file.

Reply:
Done.